json_serializable/test/integration/json_enum_example.dart

```diff
@@ -25,6 +25,19 @@ enum DayType {
 
 Iterable<String> get dayTypeEnumValues => _$DayTypeEnumMap.values;
 
+@JsonEnum(alwaysCreate: true, valueField: 'value')
```

Comment:
[probably out of scope] If we changed this to use `Symbol` then it would be limited to legal (but not necessarily correct) names for free. I don't know how many `String` arguments we currently use that correspond to Dart names. If this is the first we can consider starting with `Symbol`. If there are bunch of others we might use `String` here for consistency.

```dart
@JsonEnum(alwaysCreate: true, valueField: #value)
```

Reply:
Not out of scope! I like it!

Reply:
Nope!
https://api.dart.dev/stable/dart-core/Symbol-class.html
No way to (cleanly) get the string value out of the symbol. I'd have to do a regexp through `Symbol(fieldName)` – which feels hacky.

Also would need to teach folks about `Symbol` syntax.

Reply:
> No way to (cleanly) get the string value out of the symbol.

https://pub.dev/documentation/analyzer/latest/dart_constant_value/DartObject/toSymbolValue.html

Reply:
Yeah, but then I can't use `JsonEnum` in the generator. I need to create ANOTHER class that has a string property instead of a `Symbol` property.

I don't think it's worth it. Also not consistent w/ the other fields in the package. 🤷